### PR TITLE
Only update Protobuf and NumPy manually for compatibility/set dependabot limit

### PR DESCRIPTION
**Pull request by jcwchen**
### Description

- Only update NumPy and Protobuf manually since updating them might break compatibility
- Add proper labels
- Set threshold limit for CIs as 5 consistently

### Motivation and Context

No need to have automatic PR like https://github.com/onnx/onnx/pull/5334

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -13,6 +13,15 @@ updates:
     directory: "/" # Location of package manifests
     schedule:
       interval: "weekly"
+    ignore:
+      # Only update them manually since updating them might break compatibility
+      - dependency-name: "numpy"
+      - dependency-name: "protobuf"
+    open-pull-requests-limit: 5
```
Comment (xadupre) [inline .github/dependabot.yml:20]: Is this related to your change?

Comment (jcwchen) [reply .github/dependabot.yml:20]: Previously dependabot will automatically create unlimited PRs for PyPI package updates and this PR will set the threshold as 5 by `open-pull-requests-limit: 5`. I will mention it in PR title also.

**Review by justinchuby — APPROVED** [review]

**Review by gramalingam — APPROVED** [review]
